src/fu-device-list.c

```diff
@@ -713,6 +713,16 @@ fu_device_list_replace(FuDeviceList *self, FuDeviceItem *item, FuDevice *device)
 		fu_device_set_parent(device, parent);
 	}
 
+	/* copy the children */
+	if (fu_device_get_children(item->device)->len > 0) {
+		GPtrArray *children = fu_device_get_children(item->device);
+		for (guint i = 0; i < children->len; i++) {
+			FuDevice *child = g_ptr_array_index(children, i);
+			g_info("copying child %s to new device", fu_device_get_id(child));
+			fu_device_set_parent(child, device);
```

Comment:
Isn't it more logical to use `fu_device_add_child` on the parent?

Reply:
I've just verified that using `fu_device_add_child` would satisfy the need too, as is more appropriate to adding a child device I pushed [54007d2](https://github.com/fwupd/fwupd/commit/54007d277c914ad3309817c2d249dca3e0a76e1f) to fix it. Thanks.